### PR TITLE
Using regex to validate field value format.

### LINKED ISSUE #101: Patient Identifier type
The BI was previously  `[0-9]{9}[A-Z]{1}`, but recently it was upgraded to `[0-9]{12}[A-Z]{1}`, right now both formats are in use. 

Poc UI it takes 13 digits and a  single uppercase `[0-9]{13}[A-Z]{1}` character(1234567890123X).

### PR DESCRIPTION
**Pull request by valvijo**
Fixes #101, #103, #104

### CODE REVIEW

**Comment by larslemos** [comment]
Good job. 
We need to start using pair programming and use 2 minds and solve these issues faster.
I got stuck with these for 3 days,

File: app/registration/views/addressFields.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
                    bahmni-autocomplete source="getAddressEntryList(addressLevel.addressField)"
                    response-map="getAddressDataResults" on-select="addressFieldSelected(addressLevel.addressField)"
                    ng-change="clearFields(addressLevel.addressField)"
-                   class="form-control keyboard-in">
+                   class="form-control keyboard-in" ng-pattern="/^[a-zA-Z\s]*$/">
```
Comment (ningosi) [inline app/registration/views/addressFields.html]: I think this regex should be modified to include numbers and some other characters. If it is left to be the way it is, then things like **Distrito Municipal de KaMubukwane (antigo Distrito Urbano N 5)**  will not be allowed for **Distro**
distro

Or else those items need to be renamed to match the supplied rejex

Comment (larslemos) [inline app/registration/views/addressFields.html]: Thank you
Will update the PR

**Comment by valvijo** [comment]
There are many other situations we have to consider, for e.g: _Nacala-a-Velha, Gilé, 1º de Maio, etc_. I'm not sure about validating address format anymore.

**Pull request merged**